-----------------------------------------------------------------
Revision: 240d37f3b465527200f862ff3c0da77618ae32ae
Ancestor: 6be36979d022284283217c2a32f2b812740cbbf3
Author: datallah@pidgin.im
Date: 2007-09-14T18:26:20
Branch: im.pidgin.pidgin

Modified files:
        libpurple/proxy.c libpurple/status.c

ChangeLog: 

Apply a similar fix to the proxy prefs loading fix to the status scores.

============================================================
--- libpurple/proxy.c	395c4f16647a81960a1ac1430df4ae9fe64d05df
+++ libpurple/proxy.c	dcb73f3b56ee37fd1b03d306be342bce5167037f
@@ -1994,11 +1994,11 @@ purple_proxy_init(void)
 		proxy_pref_cb, NULL);
 
 	/* Load the initial proxy settings */
-	proxy_pref_cb("/purple/proxy/type", PURPLE_PREF_STRING, purple_prefs_get_string("/purple/proxy/type"), NULL);
-	proxy_pref_cb("/purple/proxy/host", PURPLE_PREF_STRING, purple_prefs_get_string("/purple/proxy/host"), NULL);
-	proxy_pref_cb("/purple/proxy/port", PURPLE_PREF_STRING, GINT_TO_POINTER(purple_prefs_get_int("/purple/proxy/port")), NULL);
-	proxy_pref_cb("/purple/proxy/username", PURPLE_PREF_STRING, purple_prefs_get_string("/purple/proxy/username"), NULL);
-	proxy_pref_cb("/purple/proxy/password", PURPLE_PREF_STRING, purple_prefs_get_string("/purple/proxy/password"), NULL);
+	purple_prefs_trigger_callback("/purple/proxy/type");
+	purple_prefs_trigger_callback("/purple/proxy/host");
+	purple_prefs_trigger_callback("/purple/proxy/port");
+	purple_prefs_trigger_callback("/purple/proxy/username");
+	purple_prefs_trigger_callback("/purple/proxy/password");
 }
 
 void
============================================================
--- libpurple/status.c	1eb4a9526f9b26fa096f6ff8c8205f092b126b1e
+++ libpurple/status.c	a0e82882e37ae83691ee52bc46682db953f7cf96
@@ -1668,6 +1668,14 @@ purple_status_init(void)
 	purple_prefs_connect_callback(handle, "/purple/status/scores/offline_msg",
 			score_pref_changed_cb,
 			GINT_TO_POINTER(SCORE_OFFLINE_MESSAGE));
+
+	purple_prefs_trigger_callback("/purple/status/scores/offline");
+	purple_prefs_trigger_callback("/purple/status/scores/available");
+	purple_prefs_trigger_callback("/purple/status/scores/invisible");
+	purple_prefs_trigger_callback("/purple/status/scores/away");
+	purple_prefs_trigger_callback("/purple/status/scores/extended_away");
+	purple_prefs_trigger_callback("/purple/status/scores/idle");
+	purple_prefs_trigger_callback("/purple/status/scores/offline_msg");
 }
 
 void
